### PR TITLE
GABA: Integrate CurrencyRateController

### PR DESCRIPTION
**Pull request by bitpshr**
This pull request adds `gaba/CurrencyRateController` to the extension.

### CODE REVIEW

File: app/scripts/lib/ComposableObservableStore.js [hunk 1]
```diff
@@ -40,7 +40,9 @@ class ComposableObservableStore extends ObservableStore {
   getFlatState () {
     let flatState = {}
     for (const key in this.config) {
-      flatState = { ...flatState, ...this.config[key].getState() }
+      const controller = this.config[key]
+      const state = controller.getState ? controller.getState() : controller.state
```
Comment (bitpshr) [inline app/scripts/lib/ComposableObservableStore.js]: This is a temporary patch to `ComposableObservableStore` to provide GABA API compatibility. Once we're fully on GABA, this entire module goes away and we'll be using `gaba/ComposableController` instead.

File: app/scripts/lib/getObjStructure.js [hunk 2]
```diff
@@ -7,7 +7,7 @@ module.exports = getObjStructure
 
 // {
 //   "data": {
-//     "CurrencyController": {
+//     "CurrencyRatesController": {
```
Comment (brunobar79) [inline app/scripts/lib/getObjStructure.js]: I assume this is part of a global search and replace, but in that case shouldn't be CurrencyRateController, without the **s**

**Review by brunobar79 — DISMISSED** [review]
LGTM!

**Review by brunobar79 — COMMENTED** [review]
LGTM! Ping me once the tests are passing and I'll approve again.

File: app/scripts/lib/getObjStructure.js (hunk 2 above)
Comment (bitpshr) [reply app/scripts/lib/getObjStructure.js]: Great catch, this was a manual update. Fixed.

**Review by brunobar79 — APPROVED** [review]
LGTM!

File: app/scripts/migrations/030.js [hunk 3]
```diff
@@ -0,0 +1,22 @@
+const clone = require('clone')
+
+const version = 30
```
Comment (whymarrh) [inline app/scripts/migrations/030.js]: It's worth noting that #5843 is concurrently introducing migration 30

cc @danjm

File: package.json [hunk 4]
```diff
@@ -145,6 +145,7 @@
     "fast-levenshtein": "^2.0.6",
     "file-loader": "^1.1.11",
     "fuse.js": "^3.2.0",
+    "gaba": "^1.0.0-beta.45",
```
Comment (whymarrh) [inline package.json]: We should pin this exactly so that we're not _automatically_ pulling in new prereleases

**Review by whymarrh — CHANGES_REQUESTED** [review]
Two small things:

1. I think we should pin the version of GABA we're pulling in
2. Can we add a few quick tests for the migration we're introducing?

**Comment by whymarrh** [comment]
I'm closing this for now as this will require some work to update (might be worth creating a new PR)

**Pull request closed without merging**